OpenRA.Mods.Common/Activities/Sell.cs

```diff
@@ -34,9 +35,16 @@ public Sell(Actor self, bool showTicks)
 
 		public override Activity Tick(Actor self)
 		{
-			var cost = self.GetSellValue();
+			var refund = 0;
+			long cost = self.GetSellValue();
```

Comment:
Something like the following should be fine here?
We only need to explicitly upcast one variable, and the runtime will handle the rest.
Explicitly checking for `> int.MaxValue` isn't needed because we are guaranteed that `health.HP <= health.MaxHP`.

```csharp
// Cast to long to avoid overflow when multiplying by the health
var cost = (long)self.GetSellValue();
var refund = (cost * sellableInfo.RefundPercent * (health == null ? 1 : health.HP)) / (100 * (health == null ? 1 : health.MaxHP));
playerResources.GiveCash((int)refund);
```

Reply:
Bonus points if you pull out temporary variables for `hp` and `maxHP` to make the middle line a bit more readable.

Reply:
Thanks, I learned something new!

The checking is for when someone manages to have a very big number. It is indeed a bit unnecessary and not worth the checkup.